src/Algebra/Properties/BooleanRing.agda

```diff
@@ -0,0 +1,85 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some basic properties of Rings
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles
+  using (BooleanRing; CommutativeMonoid; IdempotentCommutativeMonoid; CommutativeRing)
+
+module Algebra.Properties.BooleanRing {r₁ r₂} (R : BooleanRing r₁ r₂) where
+
+open import Function.Base using (_$_)
+
+open BooleanRing R
+open import Algebra.Definitions _≈_
+open import Algebra.Structures _≈_
+  using (IsCommutativeMonoid; IsIdempotentCommutativeMonoid; IsCommutativeRing)
+open import Relation.Binary.Reasoning.Setoid setoid
+
+------------------------------------------------------------------------
+-- Export properties of rings
+
+open import Algebra.Properties.Ring ring public
+
+------------------------------------------------------------------------
+-- Extra properties of Boolean rings
+
+xy+yx≈0 : ∀ x y → x * y + y * x ≈ 0#
+xy+yx≈0 x y = +-cancelˡ (x * x) _ _ $ +-cancelʳ (y * y) _ _ $ begin
+  x * x + ((x * y) + (y * x)) + y * y ≈⟨ +-congʳ (+-assoc _ _ _) ⟨
+  x * x + x * y + y * x + y * y       ≈⟨ +-assoc _ _ _ ⟩
+  (x * x + x * y) + (y * x + y * y)   ≈⟨ +-cong (distribˡ x x y) (distribˡ y x y) ⟨
+  x * (x + y) + y * (x + y)           ≈⟨ distribʳ (x + y) x y ⟨
+  (x + y) * (x + y)                   ≈⟨ *-idem (x + y) ⟩
+  x + y                               ≈⟨ +-congˡ (*-idem y) ⟨
+  x + y * y                           ≈⟨ +-congʳ (*-idem x) ⟨
+  x * x + y * y                       ≈⟨ +-congʳ (+-identityʳ (x * x)) ⟨
+  x * x + 0# + y * y                  ∎
+
+x+x≈0 : ∀ x → x + x ≈ 0#
+x+x≈0 x = begin
+  x + x         ≈⟨ +-cong (*-idem x) (*-idem x) ⟨
+  x * x + x * x ≈⟨ xy+yx≈0 x x ⟩
+  0#            ∎
+
+-x≈x : ∀ x → - x ≈ x
+-x≈x x = begin
+  - x ≈⟨ +-inverseˡ-unique x x (x+x≈0 x) ⟨
```

Comment:
Not worth using "equational reasoning" when it is exactly a 1-liner.

Reply:
;-) I guess it's *one* way of avoiding explicit use of `sym` (and more generally, of `trans`) but, sure...

Reply:
That's a good more general API. I've used something like that in `agda-categories`. If you go that route, I'll support it.